smarts/sstudio/genhistories.py

```diff
@@ -990,6 +1009,48 @@ def column_val_in_row(self, row, col_name: str) -> Any:
         return row[col_name]
 
 
+def import_dataset(
+    dataset_spec: types.TrafficHistoryDataset, output_path: str, overwrite: bool
+):
+    """called to pre-process (import) a TrafficHistoryDataset for use by SMARTS"""
+    if not dataset_spec.input_path:
+        print(f"skipping placeholder dataset spec '{dataset_spec.name}'.")
+        return
+    output = os.path.join(output_path, f"{dataset_spec.name}.shf")
+    if os.path.exists(output):
+        if not overwrite:
+            print("file already exists at {output}.  skipping...")
+            return
+        os.remove(output)
+    source = dataset_spec.source_type
+    if source.startswith("NGSIM"):
+        dataset = NGSIM(dataset_spec.__dict__, output)
+    elif source == "INTERACTION":
+        dataset = Interaction(dataset_spec.__dict__, output)
+    elif source == "Waymo":
```

Comment:
They have different fields in the data files.  The city scenarios (peachtree and lanker) have 6 more columns than the highway scenarios.  (See around lines 546-555 of `genhistories.py`.)

Reply:
Interesting, I assume then that peachtree and lanker are newer? 

I suppose my only issue is that this mixes in some extra implementation implications into the name rather than being just a "type".

Reply:
I agree it's not ideal.  I conceptualized the `_city` and `_highway` as "sub types".    But I might instead try to add something that determines this subtype from the number of fields in the first row of the file (and then just use `NGSIM` as the type).

Reply:
OK, I added code to infer the "subtype" from the first line of the data file, so now we can just use "NGSIM" instead of "NGSIM_city" or "NGSIM_highway".